### PR TITLE
Reapply "Bump PPM to 2024.04.2"

### PR DESCRIPTION
**Pull request by tylfin**
This reverts commit 3ada4d2e0f16492958dd88b5829641fb51ae9e32.

### CODE REVIEW

File: package-manager/.env [hunk 1]
```diff
@@ -1,5 +1,5 @@
-RSPM_VERSION=2024.04.0-20
-R_VERSION=4.2.3
+RSPM_VERSION=2024.04.2-24
```
Comment (tylfin) [inline package-manager/.env:1]:
```suggestion
RSPM_VERSION=2024.04.2-26
```

File: package-manager/Dockerfile.ubuntu2204 [hunk 2]
```diff
@@ -13,7 +13,7 @@ RUN /opt/python/${PYTHON_VERSION}/bin/python3 -m pip install --no-cache-dir buil
 RUN /opt/python/${PYTHON_VERSION_ALT}/bin/python3 -m pip install --no-cache-dir build virtualenv
 
 # Download RStudio Package Manager ---------------------------------------------#
-ARG RSPM_VERSION=2024.04.0-20
+ARG RSPM_VERSION=2024.04.2-24
```
Comment (tylfin) [inline package-manager/Dockerfile.ubuntu2204:16]:
```suggestion
ARG RSPM_VERSION=2024.04.2-26
```

File: docker-bake.hcl [hunk 3]
```diff
@@ -4,7 +4,7 @@ variable CONNECT_VERSION {
 }
 
 variable PACKAGE_MANAGER_VERSION {
-    default = "2024.04.0-20"
+    default = "2024.04.2-24"
```
Comment (tylfin) [inline docker-bake.hcl:7]:
```suggestion
    default = "2024.04.2-26"
```

File: Justfile [hunk 4]
```diff
@@ -8,7 +8,7 @@ sed_vars := if os() == "macos" { "-i ''" } else { "-i" }
 BUILDX_PATH := ""
 
 RSC_VERSION := "2024.04.1"
-RSPM_VERSION := "2024.04.0-20"
+RSPM_VERSION := "2024.04.0-26"
```
Comment (tylfin) [inline Justfile:11]:
```suggestion
RSPM_VERSION := "2024.04.0-27"
```

File: docker-bake.hcl [hunk 5]
```diff
@@ -4,7 +4,7 @@ variable CONNECT_VERSION {
 }
 
 variable PACKAGE_MANAGER_VERSION {
-    default = "2024.04.0-20"
+    default = "2024.04.2-26"
```
Comment (tylfin) [inline docker-bake.hcl:7]:
```suggestion
    default = "2024.04.2-27"
```

File: package-manager/.env [hunk 6]
```diff
@@ -1,5 +1,5 @@
-RSPM_VERSION=2024.04.0-20
-R_VERSION=4.2.3
+RSPM_VERSION=2024.04.2-26
```
Comment (tylfin) [inline package-manager/.env:1]:
```suggestion
RSPM_VERSION=2024.04.2-27
```

File: package-manager/Dockerfile.ubuntu2204 [hunk 7]
```diff
@@ -13,7 +13,7 @@ RUN /opt/python/${PYTHON_VERSION}/bin/python3 -m pip install --no-cache-dir buil
 RUN /opt/python/${PYTHON_VERSION_ALT}/bin/python3 -m pip install --no-cache-dir build virtualenv
 
 # Download RStudio Package Manager ---------------------------------------------#
-ARG RSPM_VERSION=2024.04.0-20
+ARG RSPM_VERSION=2024.04.2-26
```
Comment (tylfin) [inline package-manager/Dockerfile.ubuntu2204:16]:
```suggestion
ARG RSPM_VERSION=2024.04.2-27
```

**Comment by glin** [comment]
Is it build 27 or 28? I'm seeing the latest commit on the 2024.04 branch to be build 28, from: https://github.com/rstudio/package-manager/runs/26149261681

File: Justfile [hunk 8]
```diff
@@ -8,7 +8,7 @@ sed_vars := if os() == "macos" { "-i ''" } else { "-i" }
 BUILDX_PATH := ""
 
 RSC_VERSION := "2024.04.1"
-RSPM_VERSION := "2024.04.0-20"
+RSPM_VERSION := "2024.04.0-27"
```
Comment (tylfin) [inline Justfile:11]:
```suggestion
RSPM_VERSION := "2024.04.0-29"
```

File: docker-bake.hcl [hunk 9]
```diff
@@ -4,7 +4,7 @@ variable CONNECT_VERSION {
 }
 
 variable PACKAGE_MANAGER_VERSION {
-    default = "2024.04.0-20"
+    default = "2024.04.2-27"
```
Comment (tylfin) [inline docker-bake.hcl:7]:
```suggestion
    default = "2024.04.2-29"
```

File: package-manager/.env [hunk 10]
```diff
@@ -1,5 +1,5 @@
-RSPM_VERSION=2024.04.0-20
-R_VERSION=4.2.3
+RSPM_VERSION=2024.04.2-27
```
Comment (tylfin) [inline package-manager/.env:1]:
```suggestion
RSPM_VERSION=2024.04.2-29
```

File: package-manager/Dockerfile.ubuntu2204 [hunk 11]
```diff
@@ -13,7 +13,7 @@ RUN /opt/python/${PYTHON_VERSION}/bin/python3 -m pip install --no-cache-dir buil
 RUN /opt/python/${PYTHON_VERSION_ALT}/bin/python3 -m pip install --no-cache-dir build virtualenv
 
 # Download RStudio Package Manager ---------------------------------------------#
-ARG RSPM_VERSION=2024.04.0-20
+ARG RSPM_VERSION=2024.04.2-27
```
Comment (tylfin) [inline package-manager/Dockerfile.ubuntu2204:16]:
```suggestion
ARG RSPM_VERSION=2024.04.2-29
```

File: Justfile [hunk 12]
```diff
@@ -8,7 +8,7 @@ sed_vars := if os() == "macos" { "-i ''" } else { "-i" }
 BUILDX_PATH := ""
 
 RSC_VERSION := "2024.04.1"
-RSPM_VERSION := "2024.04.0-27"
+RSPM_VERSION := "2024.04.0-29"
```
Comment (fh-mthomson) [inline Justfile:11]:
```suggestion
RSPM_VERSION := "2024.04.2-29"
```

**Review by jonyoder — APPROVED** [review]

**Pull request merged**